tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,563 @@
+            this.refreshRate = SustainedConnectionWorker.this.spec.refreshRateMs();
+
+            // This variable is used to maintain the connection properties.
+            this.props = new Properties();
+            this.props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, SustainedConnectionWorker.this.spec.bootstrapServers());
+            this.props.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, 1);
+            WorkerUtils.addConfigsToProperties(
+                    this.props, SustainedConnectionWorker.this.spec.commonClientConf(), SustainedConnectionWorker.this.spec.producerConf());
+        }
+
+        @Override
+        public void refresh() {
+            try {
+                if (this.producer == null) {
+                    // Housekeeping to track the number of opened connections.
+                    SustainedConnectionWorker.this.totalProducerConnections.incrementAndGet();
+
+                    // Create the producer, fetch the specified topic's partitions and randomize them.
+                    this.producer = new KafkaProducer<>(this.props, new ByteArraySerializer(), new ByteArraySerializer());
+                    this.partitions = this.producer.partitionsFor(this.topicName).stream()
+                             .map(partitionInfo -> new TopicPartition(partitionInfo.topic(), partitionInfo.partition()))
+                             .collect(Collectors.toList());
+                    Collections.shuffle(this.partitions);
+                }
+
+                // Create a new iterator over the partitions if the current one doesn't exist or is exhausted.
+                if (this.partitionsIterator == null || !this.partitionsIterator.hasNext()) {
+                    this.partitionsIterator = this.partitions.iterator();
+                }
+
+                // Produce a single record and send it synchronously.
+                TopicPartition partition = this.partitionsIterator.next();
+                ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(
+                        partition.topic(), partition.partition(), keys.next(), values.next());
+                producer.send(record).get();
+
+            } catch (Throwable e) {
+                // Set the producer to be recreated on the next cycle.
+                this.closeQuietly();
+
+                // Housekeeping to track the number of opened connections and failed connection attempts.
+                SustainedConnectionWorker.this.totalProducerConnections.decrementAndGet();
+                SustainedConnectionWorker.this.totalProducerFailedConnections.incrementAndGet();
+                SustainedConnectionWorker.log.error("Error while refreshing sustained KafkaProducer connection", e.getMessage());
+
+            }
+
+            // Schedule this again and set to not in use.
+            this.nextUpdate = System.currentTimeMillis() + this.refreshRate;
```

Comment:
This is duplicated across all three classes, WDYT about moving it to the abstract?

Reply:
Done, but it leaves an open-ended potential issue of making sure refreshRate is set properly in all the subclasses.

Reply:
Yeah. I think it's good given it's a variable - we sort of expect it to be set I guess. It would be really cool if Java supported having `final abstract` variables